packages/celotool/src/lib/helm_deploy.ts

```diff
@@ -636,8 +637,9 @@ async function helmIPParameters(celoEnv: string) {
 
   const numTxNodes = parseInt(fetchEnv(envVar.TX_NODES), 10)
 
+  const apiLimit = pLimit(5)
   const txAddresses = await Promise.all(
-    range(numTxNodes).map((i) => retrieveIPAddress(`${celoEnv}-tx-nodes-${i}`))
+    range(numTxNodes).map((i) => apiLimit(() => retrieveIPAddress(`${celoEnv}-tx-nodes-${i}`)))
```

Comment:
I believe elsewhere in monorepo we use `concurrentMap` for this rather than `pLimit`.  Though, to be honest, p-limit looks like a better implementation since it picks up the tasks one by one, whereas `concurrentMap` just splits them up into sequential groups of size `concurrency`.  I'm not sure whether avoiding pLimit as a direct dependency is an explicit goal (as part of minimizing external dependencies).  Maybe @mcortesi or @hbandura would know, since I see them in the git history relating to this.

Reply:
Also, it looks like p-limit is only an indirect dependency of monorepo (I see it in some yarn.lock files, but not in any package.json).

Reply:
As the usage here is quite located and I think will be a very small difference on performance between `pLimit` and `concurrentMap`, I switched to `concurrentMap` so we do not need to add it as a direct dependency.